lib/vector-core/src/event/metric.rs

```diff
@@ -550,11 +550,17 @@ impl Metric {
                 }
             }
         };
+        let mut labels = MetricTags::new();
 
-        let labels = key
-            .labels()
-            .map(|label| (String::from(label.key()), String::from(label.value())))
-            .collect::<MetricTags>();
+        key.labels().for_each(|label| {
+            let k = String::from(label.key());
+            let v = if let Some(old_v) = labels.get(&k) {
+                format!("{}.{}", label.value(), old_v)
+            } else {
+                String::from(label.value())
+            };
```

Comment:
Sure (I should have put the PR back in draft, because the work here is not completely finished) and I did not fully assert side effects and I also need to double check that the ordering is stable.